src/main/java/seedu/meetingjio/timetables/MasterTimetable.java

```diff
@@ -166,30 +168,47 @@ public String listFree(int duration) {
         for (Timetable timetable : timetables) {
             timetable.populateBusySlots(busySlots);
         }
-        return showOutput(busySlots);
+        return showOutput(busySlots, duration);
     }
 
-    private String showOutput(int[][] busySlots) {
+    private String showOutput(int[][] busySlots, int duration) {
         String freeSlotsString = "";
+        String newEntry = "";
         boolean isStart = true;
+        int count = 0;
         for (int i = 0; i < NUM_DAYS; i++) {
             for (int j = 0; j < NUM_MINS; j++) {
-                if (busySlots[i][j] == FREE && isStart) {
-                    freeSlotsString += convertDayIntToString(i + 1);
-                    freeSlotsString += " ";
-                    freeSlotsString += convertMinsToTime(j);
-                    isStart = false;
+                if (busySlots[i][j] == FREE) {
+                    if (isStart) {
+                        newEntry += convertDayIntToString(i + 1);
+                        newEntry += " ";
+                        newEntry += convertMinsToTime(j);
+                        isStart = false;
+                    }
+                    count++;
                 }
                 if (busySlots[i][j] == BUSY && !isStart) {
-                    freeSlotsString += " ";
-                    freeSlotsString += convertMinsToTime(j);
-                    freeSlotsString += "\n";
+                    if (count >= duration * MINS_IN_1_HOUR) {
+                        newEntry += " ";
+                        newEntry += convertMinsToTime(j);
+                        newEntry += "\n";
+                        freeSlotsString += newEntry;
+                    }
+                    count = 0;
+                    newEntry = "";
                     isStart = true;
                 }
             }
         }
-        String truncatedFreeSlotsString = freeSlotsString.substring(0, freeSlotsString.length() - 1);
-        return truncatedFreeSlotsString;
+        return truncateString(freeSlotsString);
+    }
+
+    private String truncateString(String freeSlotsString) {
```

Comment:
We can have a Common class in the future to store some common methods such as truncateString and so on.